diff --git a/tablefield.module b/tablefield.module
index 26df893..3ff8e4a 100644
--- a/tablefield.module
+++ b/tablefield.module
@@ -438,7 +438,7 @@ function tablefield_field_widget_form(&$form, &$form_state, $field, $instance, $
 
   $element['tablefield'] = array(
     '#title' => $element['#title'],
-    '#description' => $help_text,
+    '#description' => filter_xss_admin($help_text),
     '#attributes' => array('id' => 'form-tablefield-' . $field['field_name'] . '-' . $delta, 'class' => array('form-tablefield')),
     '#type' => 'fieldset',
     '#tree' => TRUE,
